cli/dcos-spark/main_test.go

```diff
@@ -0,0 +1,33 @@
+package main
+
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"testing"
+
+	"github.com/mesosphere/dcos-commons/cli/client"
+	"github.com/stretchr/testify/assert"
+)
+
+func createHttpResponseWithStatus(statusCode int, filename string) http.Response {
+	status := fmt.Sprintf("%v %s", statusCode, http.StatusText(statusCode))
+	var body io.ReadCloser
+	if filename != "" {
+		content, _ := ioutil.ReadFile(filename)
+		body = ioutil.NopCloser(bytes.NewBuffer(content))
+	}
+	return http.Response{StatusCode: statusCode, Status: status, Body: body}
+}
+
+func TestUnauthorizedResponseCustomCheck(t *testing.T) {
+	response := createHttpResponseWithStatus(http.StatusUnauthorized, "testdata/responses/unauthorized.html")
+	response.Request = client.CreateServiceHTTPRequest("GET", "https://example.com/marathon/v2/apps/spark")
+
+	client.SetCustomResponseCheck(checkForUnauthorized)
+	_, err := client.CheckHTTPResponse(&response, nil)
+	assert.Contains(t, err.Error(), "Make sure you are logged in.")
```

Comment:
As we're using `response.Request.Method` and `response.Request.URL` in the error message it makes sense to assert the exact match instead of substring presence.

Reply:
Ok, changed to the exact assertion.